src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -200,22 +201,22 @@ export abstract class BaseEslintEngine extends AbstractRuleEngine {
 				// TODO: Will this break the typescript parser cwd setting?
 				const cwd = target.isDirectory ? this.baseDependencies.resolveTargetPath(target.target) : this.baseDependencies.getCurrentWorkingDirectory();
 				this.logger.trace(`Using current working directory in config as ${cwd}`);
-				const config = {cwd};
+				const config: ESLint.Options = {cwd};
 
 				target.paths = this.strategy.filterUnsupportedPaths(target.paths);
 
 				if (target.paths.length === 0) {
 					// No target files to analyze
-					this.logger.trace(`No target files to analyze from ${target.paths}`);
+					this.logger.trace(`No target files to analyze from ${JSON.stringify(target.paths)}`);
 					continue; // to the next target
 				}
 
 				// get run-config for the engine and add to config
-				Object.assign(config, deepCopy(await this.strategy.getRunConfig(engineOptions)));
+				Object.assign(config, deepCopy(await this.strategy.getRunOptions(engineOptions)));
 				// At this point, the inner `overrideConfig` property should exist. If it doesn't, we'll set it to an
 				// empty object to avoid NPEs.
-				config["overrideConfig"] = config["overrideConfig"] || {};
-				config["overrideConfig"]["rules"] = configuredRules;
+				config.overrideConfig = config.overrideConfig || {};
```

Comment:
The various new/newly-imported types let us replace the soft references with hard ones.